### DISCUSSION TITLE
How to run a command after launching via the launch_menu?

### DISCUSSION THREAD

**Discussion by tamlok** [wezterm/wezterm#4695] [Q&A] [answered]
### What Operating System(s) are you running on?

Windows

### WezTerm version

20231203-124028-e3cd2e93

### Ask your question!

I have a luanch menu item looks like:

```
    launch_menu = {
        {
            label = "EdgeTool",
            args = {"cmd", "/k", "C:\\Edge\\depot_tools\\scripts\\setup\\initEdgeEnv.cmd", "C:\\Edge"},
            cwd = "C:\\Edge\\src",
        },
    },
```

Now I want to enter the NuShell after init of the env by calling "nu". How to do this in the config?

Thanks!

**Comment by wez** [comment, marked as answer]
You'll need to send `nu\r\n` to the pane if you want it to act as though you typed that text. You can't do that from the launch menu, but you can define a key assignment that uses `action_callback` to https://wezfurlong.org/wezterm/config/lua/mux-window/spawn_tab.html and then https://wezfurlong.org/wezterm/config/lua/pane/send_text.html to send that command.

Key assignments are shown in the command palette: https://wezfurlong.org/wezterm/config/lua/keyassignment/ActivateCommandPalette.html
